build_tools/packaging/python/generate_release_index.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+"""Creates an HTML page for releases for `pip install --find-links` from a subdirectory in an S3 bucket.
```

Comment:
> Packages from the test bucket can be installed with
> 
> ```
> pip install -f https://therock-artifacts-testing.s3.us-east-2.amazonaws.com/gfx110X-dgpu/index.html rocm-sdk[libraries,devel] 
> ```

Tested on Linux in Colab: https://colab.research.google.com/gist/ScottTodd/1e489c507322857b5e64c652c8cb28ce/therock-python-install-testing.ipynb
![image](https://github.com/user-attachments/assets/4e8cd9ee-eb27-4b34-94fd-f1d8395c71db)


Works as expected.